skimage/filters/_sparse.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+from ._sparse_cy import _correlate_sparse_offsets
+from collections.abc import Iterable
+
+
+def _validate_window_size(axis_sizes):
+    """Ensure all sizes in ``axis_sizes`` are odd.
+
+    Parameters
+    ----------
+    axis_sizes : iterable of int
+
+    Raises
+    ------
+    ValueError
+        If any given axis size is even.
+    """
+    for axis_size in axis_sizes:
+        if axis_size % 2 == 0:
+            msg = ('Window size for `threshold_sauvola` or '
+                   '`threshold_niblack` must not be even on any dimension. '
+                   'Got {}'.format(axis_sizes))
+            raise ValueError(msg)
+
+
+def _to_np_mode(mode):
+    """Convert padding modes from `ndi.correlate` to `np.pad`."""
+    mode_translation_dict = dict(nearest='edge', reflect='symmetric',
+                                 mirror='reflect')
+    if mode in mode_translation_dict:
+        mode = mode_translation_dict[mode]
+    return mode
+
+
+def correlate_sparse(image, kernel, mode='reflect'):
+    """Compute valid cross-correlation of `padded_array` and `kernel`.
+
+    This function is *fast* when `kernel` is large with many zeros.
```

Comment:
```suggestion
    This function is well suited when `kernel` is large with many zeros.
```

Reply:
@rfezzani I kinda like fast better. Let's be bold, then if someone complains that their implementation is faster, we can copy it. =P

Reply:
Well, I assume that no one will complain in any case :smile:.
I just feel that "fast" is not precise, that's it....

Reply:
> I just feel that "fast" is not precise

Neither is well-suited. ;)

Reply:
We may be can find a better wording then :smile:

Reply:
I'm good with fast for now.